### PR TITLE
fix #2819: Add more unit tests for ViewLocalizer and HtmlLocalizer

### LINKED ISSUE #2819: [Localization]: Add more unit tests for ViewLocalizer and HtmlLocalizer

### PR DESCRIPTION
**Pull request by kirthik**

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs [hunk 1]
```diff
@@ -150,9 +152,168 @@ public void HtmlLocalizer_HtmlWithInvalidResourcestring_ThrowsException(string f
...
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer.GetString("John", "Doe");
+
+            // Assert
+            Assert.Equal("Hello John Doe", actualLocalizedString.Value);
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: Why isn't the expected value `"Hello John HtmlEncode[[Doe]]"`?

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs [hunk 2]
```diff
@@ -150,9 +152,168 @@ public void HtmlLocalizer_HtmlWithInvalidResourcestring_ThrowsException(string f
...
             Assert.Equal("Input string was not in a correct format.", exception.Message);
         }
+
+        [Fact]
+        public void HtmlLocalizer_GetString_ReturnsLocalizedString()
+        {
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: `CommonTestEncoder` doesn't exist. Need to get new packages, rebase, and all that stuff.

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs [hunk 3]
```diff
@@ -150,9 +152,168 @@ public void HtmlLocalizer_HtmlWithInvalidResourcestring_ThrowsException(string f
...
+        [Fact]
+        public void HtmlLocalizer_GetAllStrings_ReturnsAllLocalizedStrings()
+        {
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
+
+            // Act
+            var allLocalizedStrings = htmlLocalizer.GetAllStrings(false).ToList();
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: Name the `false` parameter (for the next reader).

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs [hunk 4]
```diff
@@ -150,9 +152,168 @@ public void HtmlLocalizer_HtmlWithInvalidResourcestring_ThrowsException(string f
...
+            Assert.Equal("World", allLocalizedStrings[0].Value);
+            Assert.Equal("Bar", allLocalizedStrings[1].Value);
+        }
     }
 
     public class TestClass
     {
     }
+
+    public class TestStringLocalizer : IStringLocalizer
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: Could these nested classes be `private`?

File: test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs [hunk 5]
```diff
@@ -77,5 +81,224 @@ public void ViewLocalizer_UseIndexerWithArguments_ReturnsLocalizedString()
...
+
+            var view = new Mock<IView>();
+            view.Setup(v => v.Path).Returns("example");
+            var viewContext = new ViewContext();
+            viewContext.View = view.Object;
+
+            viewLocalizer.Contextualize(viewContext);
+
+            // Act
+            var allLocalizedStrings = viewLocalizer.GetAllStrings(false).ToList();
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs]: Name `false` here too.

File: test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs [hunk 6]
```diff
@@ -77,5 +81,224 @@ public void ViewLocalizer_UseIndexerWithArguments_ReturnsLocalizedString()
...
+
+            // Act
+            var actualLocalizedString = viewLocalizer.WithCulture(new CultureInfo("fr"))["John"];
+
+            // Assert
+            Assert.Equal("Bonjour John", actualLocalizedString.Value);
+        }
+    }
+
+    public class TestHtmlLocalizer : IHtmlLocalizer
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs]: `private` please

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs (hunk 1 above)
Comment (kirthik) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: Encoding happens when we call `Html` method not `GetString`

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs (hunk 2 above)
Comment (kirthik) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: Thanks for pointing. I will rebase it in the next iteration.

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs (hunk 4 above)
Comment (kirthik) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: These are not nested classes. I moved them to separate files. I am using `TestStringLocalizer` in these tests and also in `ViewLocalizerTest`

File: test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs (hunk 6 above)
Comment (kirthik) [inline test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs]: I will change it.

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs (hunk 3 above)
Comment (kirthik) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: will do.

File: test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs (hunk 5 above)
Comment (kirthik) [inline test/Microsoft.AspNet.Mvc.Localization.Test/ViewLocalizerTest.cs]: will do.

**Comment by kirthik** [comment]
@dougbu updated

File: test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs]: You've added `Html()` tests but none with arguments. Think that should be covered with one test for `HtmlLocalizer` and one for `ViewLocalizer`.

**Comment by dougbu** [comment]
:shipit: with one more scenario covered in each test class.

**Comment by kirthik** [comment]
merged 3da2c35e3ecff8314205ca3ff0c46755748e2cff

**Pull request closed without merging**